### PR TITLE
ARTEMIS-2023 Fix NPE

### PR DESCRIPTION
**Pull request by michaelandrepearce**
Add test to prove NPE and compatibility with 2.6.0
Fix decoding logic to avoid NPE

### CODE REVIEW

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/client/StoreConfigTest.java [hunk 1]
```diff
@@ -289,6 +294,107 @@ public void testCreateQueue() throws Exception {
...
+      configuration.setName("np").setConnectorNames(transportConfigurations);
+
+      ByteBuffer buffer = ByteBuffer.allocate(configuration.getEncodeSize());
+      ActiveMQBuffer activeMQBuffer = new ChannelBufferWrapper(Unpooled.wrappedBuffer(buffer));
+      activeMQBuffer.clear();
+      encodeVersion260(activeMQBuffer, configuration);
+      configuration.decode(activeMQBuffer);
+   }
+
+   public void encodeVersion260(final ActiveMQBuffer buffer, final ConnectionFactoryConfigurationImpl connectionFactoryConfiguration) {
```
Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/client/StoreConfigTest.java]: I think you should use the compatibility tests for this instead. disagree with the test itself here.

Comment (michaelandrepearce) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/client/StoreConfigTest.java]: It was to just demo the introduced bug by recent merged pr. I can simply remove and leave the fix

Comment (clebertsuconic) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/client/StoreConfigTest.java]: @michaelandrepearce merged... removed your test.. added mine.. take a look please.

Comment (michaelandrepearce) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/client/StoreConfigTest.java]: Already noticed, Looks fine and dandy. Probably worth adding another version at some point maybe after 2.6.3 to the versions to be compatibility checked, just as lots of recent serialisation changes and additions

File: artemis-jms-server/src/main/java/org/apache/activemq/artemis/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java [hunk 2]
```diff
@@ -640,7 +640,7 @@ public void decode(final ActiveMQBuffer buffer) {
 
       deserializationWhiteList = BufferHelper.readNullableSimpleStringAsString(buffer);
 
-      enable1xPrefixes = buffer.readableBytes() > 0 ? buffer.readBoolean() : null;
+      enable1xPrefixes = buffer.readableBytes() > 0 ? buffer.readBoolean() : ActiveMQClient.DEFAULT_ENABLE_1X_PREFIXES;
```
Comment (jdanekrh) [inline artemis-jms-server/src/main/java/org/apache/activemq/artemis/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java]: Btw, there was an ErrorProne warning for this,

```
[17:16:53][Step 3/3] artemis-jms-server/src/main/java/org/apache/activemq/artemis/jms/server/config/impl/ConnectionFactoryConfigurationImpl.java:643: error: [NullTernary] This conditional expression may evaluate to null, which will result in an NPE when the result is unboxed.
[17:16:53][Step 3/3]       enable1xPrefixes = buffer.readableBytes() > 0 ? buffer.readBoolean() : null;
[17:16:53][Step 3/3]                                                     ^
[17:16:53][Step 3/3]     (see http://errorprone.info/bugpattern/NullTernary)
```

**Pull request closed without merging**